app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -105,6 +96,7 @@ public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, Schedule
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setGuid(BridgeUtils.generateGuid());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        plan.setDeleted(false);
```

Comment:
Unit test to make sure a schedule plan can't be created deleted?

Also, can you update the schedule plan's deleted flag? If you can update a schedule plan from deleted to not deleted, how does that impact survey and compound activity constraints?

Reply:
Created unit test in mock tests to verify you cannot create a schedule plan deleted.

Reply:
If you restore a schedule plan, it can be corrupt if you have deleted the activities you reference. We do not currently do any validation of these references when creating or updating schedule plans. I think we may have even taken it out because it was expensive. Do we want to add it back in?

Reply:
No, we don't, but we do have logic that prevents surveys and compound activities from being deleted if they are references in a schedule plan. This pull request modifies that logic to specifically exclude deleted schedule plans, which means we can circumvent this logic by deleting and then undeleting a schedule plan. This seems like an oversight.

Reply:
Updated code to reflect decision not to allow the permanent deletion of an object that's still referenced by a logically deleted object. It's OK to logically delete it though.